cdap-ranger/cdap-ranger-lookup/src/main/resources/ranger-servicedef-cdap.json

```diff
@@ -0,0 +1,279 @@
+{
+  "id": 11,
+  "name": "cdap",
+  "implClass": "co.cask.cdap.security.authorization.ranger.lookup.RangerLookupService",
+  "label": "CDAP",
+  "description": "CDAP",
+  "resources": [
+    {
+      "itemId": 1,
+      "name": "instance",
+      "type": "string",
+      "level": 10,
+      "parent": "",
+      "mandatory": true,
+      "lookupSupported": false,
+      "recursiveSupported": false,
+      "excludesSupported": false,
+      "matcher": "org.apache.ranger.plugin.resourcematcher.RangerDefaultResourceMatcher",
+      "matcherOptions": {
+        "wildCard": true,
+        "ignoreCase": true
+      },
+      "validationRegEx": "",
+      "validationMessage": "",
+      "uiHint": "",
+      "label": "CDAP Instance",
+      "description": "CDAP Instance"
+    },
+    {
+      "itemId": 2,
+      "name": "namespace",
+      "type": "string",
+      "level": 20,
+      "parent": "instance",
+      "mandatory": false,
+      "lookupSupported": true,
+      "recursiveSupported": false,
+      "excludesSupported": true,
+      "matcher": "org.apache.ranger.plugin.resourcematcher.RangerDefaultResourceMatcher",
+      "matcherOptions": {
+        "wildCard": true,
+        "ignoreCase": true
+      },
+      "validationRegEx": "",
+      "validationMessage": "",
+      "uiHint": "",
+      "label": "Namespace",
+      "description": "CDAP Namespace"
+    },
+    {
+      "itemId": 30,
+      "name": "application",
+      "type": "string",
+      "level": 30,
+      "parent": "namespace",
+      "mandatory": false,
+      "lookupSupported": true,
+      "recursiveSupported": false,
+      "excludesSupported": true,
```

Comment:
Let's try to figure this out. Otherwise how do you make the decision whether this has to be true or false?